staging/src/k8s.io/component-base/logs/api/v1/types.go

```diff
@@ -58,6 +61,23 @@ type LoggingConfiguration struct {
 	Options FormatOptions `json:"options,omitempty"`
 }
 
+type TimeOrMetaDuration metav1.Duration
+
+func (t TimeOrMetaDuration) MarshalJSON() ([]byte, error) {
+	return (metav1.Duration(t)).MarshalJSON()
+}
+
+func (t *TimeOrMetaDuration) UnmarshalJSON(b []byte) error {
```

Comment:
is this logic tested anywhere?

Reply:
I added some tests for it, see the latest changes.